src/main/java/com/rackspace/ceres/app/services/DataWriteService.java

```diff
@@ -138,4 +145,17 @@ private void cleanTags(Map<String, String> tags) {
         .retryWhen(appProperties.getRetryInsertDownsampled().build())
         .checkpoint();
   }
+
+  private Mono<?> storeMetricGroup(Metric metric) {
+    String metricGroup = metric.getMetric();
+    String metricName = metric.getMetric();
+    if (metric.getMetric().contains("_")) {
+      String[] strArr = metric.getMetric().split("_", 2);
+      metricGroup = strArr[0];
+      metricName = strArr[1];
```

Comment:
Ahhh I think I miscommunicated with you. This is correct for the metricGroup. But the metricName should still be the same thing as before. So lets look at cpu_idle as an example. The group should be `cpu`. It doesn't make sense to query for `idle`. So we should still be querying for `cpu_idle` when looking for the specific metric. At least I am pretty sure that's how it should work.

Reply:
yes, totally agreed. but I had a discussion with Cody and as per him we will get metricGroup with each and every metricName for sure. So in case of cpu_idle metricName it would be like Misc_cpu_idle, then we will store Misc as metricGroup and then cpu_idle would be metricName. 


In your example, we will search for metricGroup as "cpu" or metricName as "cpu_idle".

Reply:
Ok. Why are they doing it that way rather than adding a parameter for the metricGroup? Because it would require rebuilding the UMF object?

Reply:
they are sending a parameter for metricGroup but in umb-ceres-ingestor app, we are adding metricGroup and metricName with "_" to build metricName for us.

Reply:
So I didn't realize the metricGroup was already provided in the UMF format. We should not be doing any of this and should only be using the metricGroup provided by UMB

Reply:
https://github.com/Rackspace-Segment-Support/umb-ceres-ingestor/blob/13882967eb2429a8afaea725a6ecf0b6ac18f82a/src/main/java/com/rackspace/monplat/ingestor/services/MetricsConverter.java#L105 

here, we are creating metricName by combining metricGroup with metricName.